specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json

```diff
@@ -8998,6 +9052,35 @@
           "type": "string",
           "description": "The name of the extension."
         },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineScaleSetExtensionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResourceReadOnly"
+        }
+      ],
+      "description": "Describes a Virtual Machine Scale Set Extension."
+    },
+    "VirtualMachineScaleSetExtensionUpdate": {
+      "properties": {
+        "name": {
+          "readOnly": true,
```

Comment:
"name" is readyOnly?  (So it is not a part of request body of patch?)

Reply:
This was being mandated by ARM. Without this, it'd be

R2020 - RequiredPropertiesMissingInResourceModel 
Model definition 'VirtualMachineScaleSetExtensionUpdate' must have the properties 'name', 'id' and 'type' in its hierarchy and these properties must be marked as readonly. 
at specification/compute/resource-manager/Microsoft.Compute/stable/2019-07-01/compute.json:9108